docs/design.md

```diff
@@ -0,0 +1,206 @@
+# Design
```

Comment:
This could follow the ADR format.

Reply:
Yep, I'm thinking about restructuring the doc a bit to better fit the ADR format.

Reply:
@pjungermann The document should follow the ADR format now. It was actually very close to that already and I just needed to adjust the headings a bit.